dotCMS/src/main/java/com/dotcms/content/elasticsearch/util/DotReactiveElasticsearchClient.java

```diff
@@ -0,0 +1,874 @@
+import reactor.core.publisher.EmitterProcessor;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.FluxSink;
+import reactor.core.publisher.Mono;
+import reactor.netty.http.client.HttpClient;
+import reactor.netty.tcp.TcpClient;
+
+public class DotReactiveElasticsearchClient implements ReactiveElasticsearchClient, Indices {
+
+    private final HostProvider hostProvider;
+
+    /**
+     * Create a new {@link DotReactiveElasticsearchClient} using the given {@link HostProvider} to obtain server
+     * connections.
+     *
+     * @param hostProvider must not be {@literal null}.
+     */
+    public DotReactiveElasticsearchClient(HostProvider hostProvider) {
+
+        Assert.notNull(hostProvider, "HostProvider must not be null");
+
+        this.hostProvider = hostProvider;
+    }
+
+    /**
+     * Create a new {@link DotReactiveElasticsearchClient} aware of the given nodes in the cluster. <br />
+     * <strong>NOTE</strong> If the cluster requires authentication be sure to provide the according {@link HttpHeaders}
+     * correctly.
+     *
+     * @param headers Use {@link HttpHeaders} to provide eg. authentication data. Must not be {@literal null}.
+     * @param hosts must not be {@literal null} nor empty!
+     * @return new instance of {@link DotReactiveElasticsearchClient}.
+     */
+    public static ReactiveElasticsearchClient create(HttpHeaders headers, String... hosts) {
+
+        Assert.notNull(headers, "HttpHeaders must not be null");
+        Assert.notEmpty(hosts, "Elasticsearch Cluster needs to consist of at least one host");
+
+        ClientConfiguration clientConfiguration = ClientConfiguration.builder().connectedTo(hosts)
+                .withDefaultHeaders(headers).build();
+        return create(clientConfiguration);
+    }
+
+    /**
+     * Create a new {@link DotReactiveElasticsearchClient} given {@link ClientConfiguration}. <br />
+     * <strong>NOTE</strong> If the cluster requires authentication be sure to provide the according {@link HttpHeaders}
+     * correctly.
+     *
+     * @param clientConfiguration Client configuration. Must not be {@literal null}.
+     * @return new instance of {@link DotReactiveElasticsearchClient}.
+     */
+    public static ReactiveElasticsearchClient create(ClientConfiguration clientConfiguration) {
+
+        Assert.notNull(clientConfiguration, "ClientConfiguration must not be null");
+
+        WebClientProvider provider = getWebClientProvider(clientConfiguration);
+
+        HostProvider hostProvider = HostProvider.provider(provider,
+                clientConfiguration.getEndpoints().toArray(new InetSocketAddress[0]));
+        return new DotReactiveElasticsearchClient(hostProvider);
+    }
+
+    private static WebClientProvider getWebClientProvider(ClientConfiguration clientConfiguration) {
+
+        Duration connectTimeout = clientConfiguration.getConnectTimeout();
+        Duration soTimeout = clientConfiguration.getSocketTimeout();
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Local variable 'soTimeout' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5113278)